api/capi/include/nnstreamer.h

```diff
@@ -224,6 +230,18 @@ typedef void (*ml_pipeline_sink_cb) (const ml_tensors_data_h data, const ml_tens
  */
 typedef void (*ml_pipeline_state_cb) (ml_pipeline_state_e state, void *user_data);
 
+/**
+ * @brief Callback to execute the custom filter in NNStreamer pipelines.
+ * @since_tizen 6.0
+ * @remarks The @a in can be used only in the callback. To use outside, make a copy.
+ * @remarks The @a out can be used only in the callback. To use outside, make a copy.
+ * @param[out] in The handle of the tensor input (a single frame. tensor/tensors).
+ * @param[out] out The handle of the tensor output to be filled (a single frame. tensor/tensors).
+ * @param[out] user_data User application's private data.
+ * @return @c 0 on success. @c 1 to ignore the input data. Otherwise a negative error value.
+ */
+typedef int (*ml_custom_invoke_cb) (const ml_tensors_data_h in, ml_tensors_data_h out, void *user_data);
+
```

Comment:
add a TC or compile-time type checking macro that checks if this is equivalent to custom-easy callback.

Reply:
I implemented custom-filter registration and added testcases. Please see the changes again.